lib/cisco_node_utils/domain_name.rb

```diff
@@ -0,0 +1,54 @@
+# Domain Name provider class
+#
+# October 2015, Bryan Jen
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.join(File.dirname(__FILE__), 'node_util')
+
+module Cisco
+  # DomainName- node utility class for domain name configuration
+  class DomainName < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
```

Comment:
Agreed.  This feature should definitely be vrf aware.  The config below can coexist.
```
ip domain-name mike.cisco.com
```
or
```
vrf context blue
  ip domain-name mike.cisco.com
```